modules/nextflow/src/test/groovy/nextflow/scm/AssetManagerTest.groovy

```diff
@@ -580,4 +580,25 @@ class AssetManagerTest extends Specification {
 
     }
 
+    @Requires({System.getenv('NXF_GITHUB_ACCESS_TOKEN')})
+    def 'should download tag specified'() {
+
+        given:
+        def folder = tempDir.getRoot()
+        def token = System.getenv('NXF_GITHUB_ACCESS_TOKEN')
+        def manager = new AssetManager().build('nextflow-io/nf-test-branch', [providers: [github: [auth: token]]])
+
+        when:
+        manager.download("v0.1")
+        then:
+        folder.resolve('nextflow-io/nf-test-branch/.git').isDirectory()
+        and:
+        folder.resolve('nextflow-io/nf-test-branch/workflow.nf').text == "println 'Hello'\n"
```

Comment:
it would be much better to use a local Git repo. There's an example here

https://github.com/nextflow-io/nextflow/blob/91c1ab15e6572b2492eb85d3d1439a69eb7f6e9e/modules/nextflow/src/test/groovy/nextflow/scm/AssetManagerTest.groovy#L459-L463

Reply:
Don't think that is possible, because you need a remote git repo to replicate the issue, i.e. checkout a remote tag that hasn't been pulled locally yet

Reply:
Fair enough